### PR TITLE
Search country in country dropdown, enable/disable default country selection, maxLength and tooltip on selected country, dynamic change in preferred countries

### PR DESCRIPTION
**Pull request by snmhinge**
An input text box is added in the country dropdown to enable search country, added input properties to search country i.e. searchCountryFlag and serchcountryField. Added maxLength input property to explicitly provide a limit to the input phone number. Added tooltipField input property to enable tooltip selected country based on name or iso2. Added defaultFirstCountrySelected input field to disable default select country from countries dropdown. Updated readme document.

### CODE REVIEW

**Comment by snmhinge** [comment]
@pasevin Please check the PR. I've resolved all the bugs in my previous PR. And also added dynamic change in preferred countries.

File: projects/ngx-intl-tel-input/src/lib/ngx-intl-tel-input.validator.ts [hunk 1]
```diff
@@ -2,25 +2,30 @@ import { FormControl } from '@angular/forms';
 import * as lpn from 'google-libphonenumber';
 
 export const phoneNumberValidator = (control: FormControl) => {
-	const isRequired = control.errors && control.errors.required === true;
-	const error = { validatePhoneNumber: { valid: false } };
-	let number: lpn.PhoneNumber;
+	const isCheckValidation = document.getElementById('phone').getAttribute('validation');
```
Comment (pasevin) [inline projects/ngx-intl-tel-input/src/lib/ngx-intl-tel-input.validator.ts]: Can you avoid using ```document.getElementById``` and instead do it Angular way?

File: projects/ngx-intl-tel-input/src/lib/ngx-intl-tel-input.component.ts [hunk 2]
```diff
@@ -33,26 +33,36 @@ export class NgxIntlTelInputComponent implements OnInit {
...
+		name: '',
+		placeHolder: '',
+		priority: 0
+	};
 
 	phoneNumber = '';
 	allCountries: Array<Country> = [];
 	preferredCountriesInDropDown: Array<Country> = [];
-	selectedCountry: Country;
+	// selectedCountry: Country;
```
Comment (pasevin) [inline projects/ngx-intl-tel-input/src/lib/ngx-intl-tel-input.component.ts]: Just delete this.

File: README.md [hunk 3]
```diff
@@ -73,15 +73,16 @@ Or this [Stackblitz Demo](https://stackblitz.com/edit/ngx-intl-tel-input-demo).
...
-  [preferredCountries]="['us', 'gb']"
+  [preferredCountries]="preferredCountries"
   [onlyCountries]="['us', 'gb', 'es']"
   [enableAutoCountrySelect]="true"
   [enablePlaceholder]="true"
   [searchCountryFlag]="true"
-  [searchCountryFeild]="'all'"
-  [defaultFirstCountrySelected]="true"
-  [maxLength]=""
+  [searchCountryFeild]="['iso2','name']'"
```
Comment (pasevin) [inline README.md]: typo: ```searchCountryFeild```

File: README.md [hunk 4]
```diff
@@ -98,10 +99,11 @@ Or this [Stackblitz Demo](https://stackblitz.com/edit/ngx-intl-tel-input-demo).
 | enableAutoCountrySelect       | ```boolean```          | ```false```        | Toggle automatic country (flag) selection based on user input.                      |
 | enablePlaceholder             | ```boolean```          | ```true```         | Input placeholder text, which addapts to the country selected.                      |
 | searchCountryFlag             | ```boolean```          | ```false```         | toggle if you want to search country usin using inout text      |
-| searchCountryFeild             | ```string```          | ``````         | fields(name, iso2, dialCode, all) based on which search country. 'all' is used when you want to search country based on all(name, iso2, dialCode) fields. |
+| searchCountryFeild             | ```string[]```          | ```[]```         | List of fields having options (name, iso2, dialCode) based on which search country. 'all' is used when you want to search country based on all(name, iso2, dialCode) fields. |
```
Comment (pasevin) [inline README.md]: typo: ```searchCountryFeild```

**Comment by pasevin** [comment]
Thank you! I will push some of my updates and fixes on top of this. I found some small bugs.

**Pull request merged**